src/components/istio_ambient_requirer_component.py

```diff
@@ -0,0 +1,90 @@
+from charmed_kubeflow_chisme.components import Component
+from charms.istio_beacon_k8s.v0.service_mesh import ServiceMeshConsumer
+from charms.istio_ingress_k8s.v0.istio_ingress_route import (
+    BackendRef,
+    HTTPPathMatch,
+    HTTPPathMatchType,
+    HTTPRoute,
+    HTTPRouteMatch,
+    IstioIngressRouteConfig,
+    IstioIngressRouteRequirer,
+    Listener,
+    PathModifier,
+    PathModifierType,
+    ProtocolType,
+    URLRewriteFilter,
+    URLRewriteSpec,
+)
+from ops import ActiveStatus, BlockedStatus
+
+
+class AmbientIngressRequirerComponent(Component):
+    """Component to manage the relations to Istio ambient."""
+
+    def __init__(
+        self,
+        *args,
+        service_name: str,
+        port: int,
+        path_prefix: str,
+        url_rewrite: str,
+        relation_name: str = "istio-ingress-route",
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+
+        self.relation_name = relation_name
+        self.service_name = service_name
+        self.port = port
+        self.path_prefix = path_prefix
+        self.url_rewrite = url_rewrite
+
+        self.ingress = IstioIngressRouteRequirer(
+            self._charm,
+            relation_name=self.relation_name,
+        )
+
+        self._config = self._get_ingress_config()
+
+        self._mesh = ServiceMeshConsumer(self._charm)
+
+    def _get_ingress_config(self):
+        # Define listeners - names are auto-generated by the charm
+        http_listener = Listener(port=80, protocol=ProtocolType.HTTP)
+
+        config = IstioIngressRouteConfig(
+            model=self._charm.model.name,  # Requirer's namespace where services live
+            listeners=[http_listener],
+            http_routes=[
+                HTTPRoute(
+                    name="http-route",
+                    listener=http_listener,
+                    matches=[
+                        HTTPRouteMatch(
+                            path=HTTPPathMatch(
+                                type=HTTPPathMatchType.PathPrefix, value=self.path_prefix
+                            ),
+                        )
+                    ],
```

Comment:
suggestion(blocking): It looks like they have changed the API for defining the matches. Let's fetch the latest library and rewrite this part

I noticed this during the mlflow PR. The code should look something like this:
https://github.com/canonical/mlflow-operator/pull/385/files#diff-b9ed39bbc9c0387bd3e07da31d13373745534a1cd723d3e292c73496b12e307cR183
```python
matches=[HTTPRouteMatch(path=HTTPPathMatch(value="/mlflow/"))],
```

Reply:
good catch, 8dcffb8